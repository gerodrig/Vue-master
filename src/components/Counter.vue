<template>
    <div class="mx-auto text-center debug">
        <h1 class="text-4xl">Counter</h1>
        <p data-testid="counter" class="my-5 text-2xl"> {{ count }} <sup>2</sup> = {{ squared }}</p>
        <div class="buttons">
        <button data-testid="decrement" class="px-4 py-2 font-bold text-white bg-yellow-500 rounded hover:bg-yellow-700" @click="decrement">-1</button>
        <button data-testid="increment" class="px-4 py-2 ml-5 font-bold text-white bg-green-500 rounded hover:bg-green-700" @click="increment">+1</button>
        </div>
    </div>
</template>


<script lang="ts">
    export default {
        name: 'Counter',
        data() {
            return {
                count: 0
            }
        },
        methods: {
            increment() {
                this.count++
            },
            decrement() {
                this.count--
            }
        },
        computed: {
            squared() {
                return this.count * this.count
            }
        },
    }
</script>