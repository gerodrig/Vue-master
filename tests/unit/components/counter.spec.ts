import { describe, it, expect, beforeEach } from 'vitest';

import { shallowMount } from "@vue/test-utils";


import Counter from '../../../src/components/Counter.vue'


describe('Counter Component', async () => {

    let wrapper: any;

    beforeEach(() => {
        wrapper = shallowMount(Counter);
    });
    
        it('must match the snapshot', () => {
    
            //arrange
            // const wrapper = shallowMount(Counter);

            //console.log(wrapper.html());

            //assert
            expect(wrapper.html()).toMatchSnapshot();
    
        });


        it('h1 must have the text "Counter"', () => {

            //arrange
            // const wrapper = shallowMount(Counter);

            //assert
            expect(wrapper.find('h1').text()).toBe('Counter');
        });

        it('default value in p must be 0 2 = 0"', () => {
                
                //arrange
                // const wrapper = shallowMount(Counter);

                const value = wrapper.find('[data-testid="counter"]').text();

    
                //assert
                expect(value).toBe('0 2 = 0');
        });

        it('must increment the counter in 1 when button is clicked', async () => {

            //arrange
            // const wrapper = shallowMount(Counter);

            const [decrement, increment] = wrapper.findAll('button');

            //act
            await increment.trigger('click');

            //click twice decrement
            await decrement.trigger('click');
            await decrement.trigger('click');

            const value = wrapper.find('[data-testid="counter"]').text();

            //assert
            expect(value).toBe('-1 2 = 1');
        });

        it('must define the default value of the counter', () => {

            //get count default value
            const defaultValue = wrapper.vm.count

            //assert
            expect(defaultValue).toBe(0);
        });
});