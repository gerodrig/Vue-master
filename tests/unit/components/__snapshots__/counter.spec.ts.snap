// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Counter Component > must match the snapshot 1`] = `
"<div class=\\"mx-auto text-center debug\\">
  <h1 class=\\"text-4xl\\">Counter</h1>
  <p data-testid=\\"counter\\" class=\\"my-5 text-2xl\\">0 <sup>2</sup> = 0</p>
  <div class=\\"buttons\\"><button data-testid=\\"decrement\\" class=\\"px-4 py-2 font-bold text-white bg-yellow-500 rounded hover:bg-yellow-700\\">-1</button><button data-testid=\\"increment\\" class=\\"px-4 py-2 ml-5 font-bold text-white bg-green-500 rounded hover:bg-green-700\\">+1</button></div>
</div>"
`;
